integration-libs/opf/checkout/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.service.ts

```diff
@@ -0,0 +1,270 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ActiveCartService } from '@spartacus/cart/base/core';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  HttpErrorModel,
+  HttpResponseStatus,
+  RoutingService,
+  UserIdService,
+  backOff,
+} from '@spartacus/core';
+import {
+  isAuthorizationError,
+  opfAuthorizationErrorRetry,
+} from '@spartacus/opf/base/core';
+import {
+  OpfOrderFacade,
+  OpfOtpFacade,
+  OpfResourceLoaderService,
+  OpfService,
+} from '@spartacus/opf/base/root';
+import {
+  OPF_PAYMENT_AND_REVIEW_SEMANTIC_ROUTE,
+  OpfCheckoutFacade,
+  OpfPaymentMethodType,
+  OpfRenderPaymentMethodEvent,
+  PaymentPattern,
+  PaymentSessionData,
+} from '@spartacus/opf/checkout/root';
+import {
+  BehaviorSubject,
+  Observable,
+  combineLatest,
+  of,
+  throwError,
+} from 'rxjs';
+import { catchError, filter, map, switchMap, take, tap } from 'rxjs/operators';
+
+@Injectable()
+export class OpfCheckoutPaymentWrapperService {
+  protected lastPaymentOptionId?: number;
+
+  protected activeCartId?: string;
+
+  protected renderPaymentMethodEvent$ =
+    new BehaviorSubject<OpfRenderPaymentMethodEvent>({
+      isLoading: false,
+      isError: false,
+    });
+
+  constructor(
+    protected opfCheckoutService: OpfCheckoutFacade,
+    protected opfOtpService: OpfOtpFacade,
+    protected opfResourceLoaderService: OpfResourceLoaderService,
+    protected userIdService: UserIdService,
+    protected activeCartService: ActiveCartService,
+    protected routingService: RoutingService,
+    protected globalMessageService: GlobalMessageService,
+    protected opfOrderFacade: OpfOrderFacade,
+    protected opfService: OpfService
+  ) {}
+
+  protected executeScriptFromHtml(html: string): void {
+    /**
+     * Verify first if customer is still on the payment and review page.
+     * Then execute script extracted from HTML to render payment provider gateway.
+     */
+    this.routingService
+      .getRouterState()
+      .pipe(
+        take(1),
+        filter(
+          (route) =>
+            route.state.semanticRoute === OPF_PAYMENT_AND_REVIEW_SEMANTIC_ROUTE
+        )
+      )
+      .subscribe(() => {
+        setTimeout(() => {
```

Comment:
[MEDIUM]

The use of `setTimeout` seems to be a workaround, but it is not explained in the code comments WHY it is necessary.

Moreover, are we _really_ sure that the issue will be resolved in the next macrotask (`setTimeout(,0)`)? To be more bullet-proof, shouldn't we wait for a specific application's event instead?